backend/src/database/databaseConnection.js

```diff
@@ -0,0 +1,211 @@
+   * Checks the Knex connection, the database schema, and Objection models.
+   * @returns {boolean} True if successful, otherwise false.
+   */
+  async checkAll() {
+    const connectOk = await this.checkConnection();
+    const schemaOk = await this.checkSchema();
+    //const modelsOk = this.checkModel();
+
+    console.log(`Connect OK: ${connectOk}, Schema OK: ${schemaOk}`, { function: 'checkAll' });
+    this._connected = connectOk && schemaOk;
+    return this._connected;
+  }
+
+  /**
+   * @function checkConnection
+   * Checks the current knex connection to Postgres.
+   * If the connected DB is in read-only mode, transaction_read_only will not be off.
+   * @returns {boolean} True if successful, otherwise false.
+   */
+  async checkConnection(no_bootstrap = false) {
+    try {
+      const data = await this.knex.raw('show transaction_read_only');
+      const result = data && data.rows && data.rows[0].transaction_read_only === 'off';
+      if (result) {
+        console.log('Database connection ok', { function: 'checkConnection' });
+      } else {
+        console.log('Database connection is read-only', { function: 'checkConnection' });
+      }
+      return result;
+    } catch (err) {
+      console.log(`Error with database connection: ${err.message}`, { function: 'checkConnection' });
+
+      // If the database does not exist, and auto-deploy is enabled, attempt to bootstrap the database.
+      if (!no_bootstrap && process.env.DATABASE_AUTO_DEPLOY === '1' && err.message === `database "${knexConfig.connection.database}" does not exist`) {
+        console.log(`db bootstrap: database "${knexConfig.connection.database}" not found. attempting to auto-bootstrap it...`);
+        if (await this.bootstrapDatabase(knexConfig.connection.database)) {
+          // If the bootstrapping seemed successful, call ourself to see if we can connect properly now, but don't try to bootstrap again.
+          return this.checkConnection(true);
+        }
+      }
+
+      return false;
+    }
+  }
+
+  /**
+   * @function checkSchema
+   * Queries the knex connection to check for the existence of the expected schema tables.
+   * @returns {boolean} True if schema is ok, otherwise false.
+   */
+  checkSchema() {
+    const tables = ['migrations'];
+    try {
+      return Promise
+        .all(tables.map(table => this._knex.schema.hasTable(table)))
+        .then(exists => exists.every(x => x))
+        .then(result => {
+          if (result) {
+            console.log('Database schema ok', { function: 'checkSchema' });
+          }
+          return result;
+        });
+    } catch (err) {
+      console.log(`Error with database schema: ${err.message}`, { error: err, function: 'checkSchema' });
+      return false;
+    }
+  }
+
+  /**
+   * @function checkModel
+   * Attaches the Objection model to the existing knex connection.
+   * @returns {boolean} True if successful, otherwise false.
+   */
+  /*
+  checkModel() {
```

Comment:
Just wondering why this code is commented out.  Is it suppose to have a purpose, or just testing something.